### PR TITLE
Implemented SSAO and "Recompile Shaders" Button on Editor.

### PR DESCRIPTION
**Pull request by baransrc**
To test the SSAO, see the "Ambient Occlusion" settings on the Render pane of Configurations tab. Also check the new added mode on deferred renderer settings "SSAO Color" that displays the texture of SSAO. Changes to the SSAO Configurations are saved through editor preferences, "he.cfg" so if we tweak the values for SSAO for the final build should make sure that we have the correct SSAO Configuration on he.cfg.

NOTE: Current implementation of SSAO suffers from a "white banding" issue on certain angles of the camera. The issue is reproducable only in engine mode as the camera has a free control. In gameplay we don't have the issue. I've notified our teacher Carlos about the issue, and waiting for an answer, at the time of this comment. If we can find a fix, I will be doing the fix on a separate branch.

--------
To test the Shader rebuild, you can make changes on shader(s) and see the results applied immediately by clicking on Shaders->Recompile from the header bar. The changes however, if have errors, engine will crash. (Still better than before, it really helped me during the implementation of SSAO, and took 5 minutes to implement)

### CODE REVIEW

**Review by Bernatmago — APPROVED** [review]
Working like a charm!

File: Game/shaders/fragment_ssao.glsl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    vec3 random_vec = normalize(texture(texture_noise, texture_coords * noise_scale).xyz);
+
+    // Create TBN change-of-basis matrix. From tangent-space to view-space:
+    vec3 tangent = normalize((random_vec - fragment_normal) * dot(random_vec, fragment_normal));
+    vec3 bitangent = normalize(cross(fragment_normal, tangent));
+    mat3 tbn = mat3(tangent, bitangent, fragment_normal);
+
+    // iterate over the sample kernel and calculate occlusion factor
+    float occlusion = 0.0;
+    float occluding_points = 0.0;
```
Comment (baransrc) [inline Game/shaders/fragment_ssao.glsl:40]:
```suggestion
```

File: Source/src/core/rendering/GBuffer.cpp [hunk 2]
```diff
@@ -240,3 +240,26 @@ void Hachiko::GBuffer::UnbindFogTextures() const
     glActiveTexture(GL_TEXTURE3);
     glBindTexture(GL_TEXTURE_2D, 0);
 }
+
+void Hachiko::GBuffer::BindSSAOTextures() const
+{
+    //// Bind diffuse texture:
+    //glActiveTexture(GL_TEXTURE0);
+    //glBindTexture(GL_TEXTURE_2D, _diffuse_texture);
```
Comment (baransrc) [inline Source/src/core/rendering/GBuffer.cpp:248]:
```suggestion
```

File: Source/src/core/rendering/ShadowMappingProperties.h [hunk 3]
```diff
@@ -54,14 +54,15 @@ class ShadowMappingProperties
...
-    float _gaussian_filter_blur_amount = 0.0f;
     float _min_variance = 0.0f;
     float _light_bleeding_reduction_amount = 0.0f;
     float _bias = 0.0f;
     float _exponent = 0.0f;
-    BlurPixelSize::Type _gaussian_blur_size = BlurPixelSize::Type::Gaussian5x5;
-    float _gaussian_blur_sigma = 0.0f;
+    //float _gaussian_filter_blur_amount = 0.0f;
+    //BlurPixelSize::Type _gaussian_blur_size = BlurPixelSize::Type::Gaussian5x5;
+    //float _gaussian_blur_sigma = 0.0f;
```
Comment (baransrc) [inline Source/src/core/rendering/ShadowMappingProperties.h:65]:
```suggestion
```

**Review by VBoschMatas — APPROVED** [review]
Looks clean af

**Review by monikamnc — APPROVED** [review]
GREEEAT MA DUUUDE!!!!!!!

**Pull request merged**